trin-core/src/utp/stream.rs

```diff
@@ -210,73 +249,65 @@ impl UtpListener {
-                                tokio::spawn(async move {
-                                    debug!(
-                                        "Sending content data via uTP with len: {}",
-                                        content_data.len()
-                                    );
-                                    // send the content to the requestor over a uTP stream
-                                    if let Err(msg) = conn
-                                        .send_to(
-                                            &UtpMessage::new(content_data.as_ssz_bytes()).encode()
-                                                [..],
-                                        )
-                                        .await
-                                    {
-                                        error!("Error sending content {msg}");
-                                    } else {
-                                        // Close uTP connection
-                                        if let Err(msg) = conn.close().await {
-                                            error!("Unable to close uTP connection!: {msg}")
-                                        }
-                                    };
-                                });
+                                debug!(
+                                    "Sending content data via uTP with len: {}",
+                                    content_data.len()
+                                );
+                                // send the content to the requestor over a uTP stream
+                                if let Err(msg) = conn
+                                    .send_to(
+                                        &UtpMessage::new(content_data.as_ssz_bytes()).encode()[..],
+                                    )
+                                    .await
+                                {
+                                    error!("Error sending content {msg}");
+                                } else {
+                                    // Close uTP connection
+                                    if let Err(msg) = conn.close().await {
+                                        error!("Unable to close uTP connection!: {msg}")
+                                    }
+                                };
                             }
                         } else {
-                            warn!("Query requested an unknown ENR");
+                            warn!("Received SYN packet for an unknown active uTP stream");
                         }
                     }
                     // Receive DATA and FIN packets
                     PacketType::Data => {
                         if let Some(conn) = self
                             .utp_connections
-                            .get_mut(&ConnectionKey::new(*node_id, connection_id))
+                            .get_mut(&ConnectionKey::new(*node_id, connection_id - 1))
```

Comment:
Why the minus one here?

Reply:
This is because we are expecting to receive a SYN packet with the random connection id (conn_id) generated by the acceptor node (us) and this is the value we are using to create `ConnectionKey` to add the stream to the active connections hashmap (when handling OFFER message for example).

However, when we receive DATA or FIN packets, we are expecting a packet to have `conn_id + 1` value (according to uTP [specs](https://www.bittorrent.org/beps/bep_0029.html)). So if we want to check if the packet is part of an active uTP stream, when handling those packets, we have to generate a `ConnectionKey` using the` packet.connection_id - 1`.

The good point here is that we should handle the overflow and use `wrapped_sub(1)`

Reply:
Hmm I see. Is there a test case that we can add for this logic?

Reply:
Added an integration test for the uTP listener events that catches such types of changes.